qiita_db/study.py

```diff
@@ -347,6 +348,56 @@ def create(cls, owner, title, efo, info, investigation=None):
 
         return cls(study_id)
 
+    @classmethod
+    def delete(cls, id_):
+        r"""Deletes the study from the database
+
+        Parameters
+        ----------
+        id_ : integer
+            The object identifier
+
+        Raises
+        ------
+        QiitaDBError
+            If the sample_(id_) table exists means a sample template exists
+        """
+        cls._check_subclass()
+
+        conn_handler = SQLConnectionHandler()
+        if exists_table('sample_%d' % id_, conn_handler):
```

Comment:
The way it works right now is that you need a raw data to have a preprocessed data, a preprocessed data to have a processed data, which I think is fine and give us the possibility of "play" with the options. For example, when we allow to upload a process data we can link it to the same default raw data for faster searches.